app/objects/c_ability.py

```diff
@@ -18,89 +17,54 @@ class AbilitySchema(ma.Schema):
     technique_id = ma.fields.String(missing=None)
     name = ma.fields.String(missing=None)
     description = ma.fields.String(missing=None)
-    cleanup = ma.fields.List(ma.fields.String(), missing=None)
-    executor = ma.fields.String(missing=None)
-    platform = ma.fields.String(missing=None)
-    payloads = ma.fields.List(ma.fields.String(), missing=None)
-    uploads = ma.fields.List(ma.fields.String(), missing=None)
-    parsers = ma.fields.List(ma.fields.Nested(ParserSchema), missing=None)
-    requirements = ma.fields.List(ma.fields.Nested(RequirementSchema), missing=None)
+    executors = ma.fields.List(ma.fields.Nested(ExecutorSchema()), missing=None)
+    requirements = ma.fields.List(ma.fields.Nested(RequirementSchema()), missing=None)
     privilege = ma.fields.String(missing=None)
-    timeout = ma.fields.Int(missing=60)
     repeatable = ma.fields.Bool(missing=None)
-    language = ma.fields.String(missing=None)
-    code = ma.fields.String(missing=None)
-    build_target = ma.fields.String(missing=None)
-    variations = ma.fields.List(ma.fields.Nested(VariationSchema), missing=None)
     buckets = ma.fields.List(ma.fields.String(), missing=None)
     additional_info = ma.fields.Dict(keys=ma.fields.String(), values=ma.fields.String())
     access = ma.fields.Nested(AccessSchema, missing=None)
-    test = ma.fields.String(missing=None)
     singleton = ma.fields.Bool(missing=None)
 
     @ma.post_load
     def build_ability(self, data, **_):
-        if 'technique_name' in data:
-            data['technique'] = data.pop('technique_name')
+        if 'technique' in data:
+            data['technique_name'] = data.pop('technique')
         return Ability(**data)
 
 
 class Ability(FirstClassObjectInterface, BaseObject):
 
     schema = AbilitySchema()
-    # may need to fix for id=self.unique
-    display_schema = AbilitySchema(exclude=['repeatable', 'language', 'code', 'build_target', 'singleton'])
+    display_schema = AbilitySchema()
 
-    RESERVED = dict(payload='#{payload}')
     HOOKS = dict()
 
-    @property
-    def test(self):
-        return self.replace_app_props(self._test)
-
-    @test.setter
-    def test(self, cmd):
-        self._test = self.encode_string(cmd)
-
     @property
     def unique(self):
-        return '%s%s%s' % (self.ability_id, self.platform, self.executor)
+        return self.ability_id
 
     @property
-    def raw_command(self):
-        return self.decode_bytes(self._test) if self._test else ""
-
-    @classmethod
-    def is_global_variable(cls, variable):
```

Comment:
I've moved this to the executor, to be next to the command. Although it could probably go back to the ability, too.

Reply:
I'm honestly starting to think it might be worth putting it in the base object that everything inherits. I'm not certain it necessarily needs to be here over it being in executor, just wanted to make sure that it being removed doesn't break anything planning side.

Reply:
Could this be done in a separate PR after this one is committed?